nebula_ros/src/continental/continental_ars548_decoder_wrapper.cpp

```diff
@@ -338,10 +360,289 @@ void ContinentalARS548DecoderWrapper::packets_callback(
+    autoware_object.existence_probability =
+      continental_object.existence_probability / continental_ns::raw_prob_norm;
+
+    // Position
+    // There are 9 possible reference points. In the case of an invalid reference point, we fall
+    // back to the center
+    const int reference_index = std::min<int>(continental_object.position_reference, 8);
+    const double & yaw = continental_object.orientation;
+    autoware_object.position = reference_point_to_center(
+      continental_object.position, yaw, continental_object.shape_length_edge_mean,
+      continental_object.shape_width_edge_mean, reference_index);
+
+    autoware_object.velocity = continental_object.absolute_velocity;
+    autoware_object.acceleration = continental_object.absolute_acceleration;
+    autoware_object.size.x = continental_object.shape_length_edge_mean;
+    autoware_object.size.y = continental_object.shape_width_edge_mean;
+    autoware_object.size.z = 1.f;
+
+    RadarClassification classification;
+    autoware_object.classifications.reserve(10);
+
+    classification.label = RadarClassification::UNKNOWN;
+    classification.probability =
+      continental_object.classification_unknown / continental_ns::raw_prob_norm;
+    autoware_object.classifications.push_back(classification);
+
+    classification.label = RadarClassification::CAR;
+    classification.probability =
+      continental_object.classification_car / continental_ns::raw_prob_norm;
+    autoware_object.classifications.push_back(classification);
+
+    classification.label = RadarClassification::TRUCK;
+    classification.probability =
+      continental_object.classification_truck / continental_ns::raw_prob_norm;
+    autoware_object.classifications.push_back(classification);
+
+    classification.label = RadarClassification::MOTORCYCLE;
+    classification.probability =
+      continental_object.classification_motorcycle / continental_ns::raw_prob_norm;
+    autoware_object.classifications.push_back(classification);
+
+    classification.label = RadarClassification::BICYCLE;
+    classification.probability =
+      continental_object.classification_bicycle / continental_ns::raw_prob_norm;
+    autoware_object.classifications.push_back(classification);
+
+    classification.label = RadarClassification::PEDESTRIAN;
+    classification.probability =
+      continental_object.classification_pedestrian / continental_ns::raw_prob_norm;
+    autoware_object.classifications.push_back(classification);
+
+    classification.label = RadarClassification::ANIMAL;
+    classification.probability =
+      continental_object.classification_animal / continental_ns::raw_prob_norm;
+    autoware_object.classifications.push_back(classification);
+
+    classification.label = RadarClassification::HAZARD;
+    classification.probability =
+      continental_object.classification_hazard / continental_ns::raw_prob_norm;
+    autoware_object.classifications.push_back(classification);
+
```

Comment:
I think these warrant a helper function like `normalize_probability(continental_object.xxxx)` for readability.

Come to think of it, the fields of `continental_object` should probably be prefixed with `raw_prob_` to make clear it's not a usable-as-is value, and that it's related to `raw_prob_norm`.

Reply:
Addressed in https://github.com/tier4/nebula/pull/284/commits/c4d3ca83df118a1e7bb87b6f00a80beac9286203
Since this changed the tests, the reported lines changed exploded :sweat_smile: